src/vis/layers-factory.js

```diff
@@ -95,12 +95,7 @@ var LAYER_CONSTRUCTORS = {
   },
 
   cartodb: function (attrs, options) {
-    // TODO: Once https://github.com/CartoDB/cartodb/issues/12885 is merged,
-    // we should make 'source' attribute required again and make sure it's
-    // always populated:
-    // checkProperties(attrs, ['source', 'cartocss']);
-    // attrs.source = CartoDBLayer.getLayerSourceFromAttrs(attrs, options.vis.analysis);
-    checkProperties(attrs, ['cartocss']);
+    checkProperties(attrs, ['cartocss', 'source']);
```

Comment:
https://github.com/CartoDB/cartodb/issues/12885 is not closed yet so we cannot make "source" required.